micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetricsTest.java

```diff
@@ -65,13 +65,18 @@ void noJvmImplementationSpecificApiSignatures() {
     }
 
     @Test
-    void metersAreBound() {
-        assertThat(registry.find("jvm.gc.live.data.size").gauge()).isNotNull();
-        assertThat(registry.find("jvm.gc.memory.allocated").counter()).isNotNull();
-        assertThat(registry.find("jvm.gc.max.data.size").gauge().value()).isGreaterThan(0);
+    void gcMetricsAvailableAfterGc() {
+        System.gc();
+        await().timeout(200, TimeUnit.MILLISECONDS).alias("NotificationListener takes time after GC")
+                .untilAsserted(() ->
+                        assertThat(registry.find("jvm.gc.live.data.size").gauge().value()).isPositive());
```

Comment:
By forcing a GC and waiting for the notification listener to update metrics, this test does a much better job of ensuring expected metrics are available for the configured collector.